admin/app/components/solidus_admin/main_nav/component.rb

```diff
@@ -1,27 +1,18 @@
 # frozen_string_literal: true
 
-module SolidusAdmin
-  # Renders the main navigation of Solidus Admin.
-  class MainNav::Component < BaseComponent
-    include Import[
-      main_nav_item_component: "components.main_nav_item.component",
-      items: "main_nav_items"
-    ]
+# Renders the main navigation of Solidus Admin.
+class SolidusAdmin::MainNav::Component < SolidusAdmin::BaseComponent
+  include Import[items: "main_nav_items"]
 
-    erb_template <<~ERB
-      <nav>
-        <%=
-          render main_nav_item_component.with_collection(
-            sorted_items
-          )
-        %>
-      </nav>
-    ERB
+  erb_template <<~ERB
+    <nav>
+      <%= render component('main_nav/item').with_collection(sorted_items) %>
```

Comment:
Caution! By calling the component from here, we're breaking the possibility of injecting new components on initialization. It'd be nice for a host application to be able to switch which nested component a parent component renders by just changing the initializer and without the need to touch the parent template. That would help for situations where more than one parent can use a nested component, and you only want to do the substitution in one place. In general, it would make the components less coupled and make testing easier and faster.

I'll prioritize working on the "Readme" about component extensibility to clarify the intentions behind using those patterns here. I'll try to have it ready by the beginning of next week :crossed_fingers:

Reply:
I'll move the commit out of this PR as it probably needs it's own space 👍